### PR TITLE
Steam find

### PR DESCRIPTION
**Pull request by elken**
Discussion on #33

### CODE REVIEW

**Comment by elken** [comment]
:cry:

**Comment by lolzballs** [comment]
Ugh, more merge conflicts now :/

**Comment by elken** [comment]
Have fun fixing them!

Sorry I can't help fix my own shit xD

**Comment by lolzballs** [comment]
Can you revert that commit? After #37 is merged, you can merge the dev branch into your steam one, and everything will be fine!

**Comment by elken** [comment]
DId that do it?

EDIT: yes

**Comment by lolzballs** [comment]
Yeah, now you should be able to merge the latest changes into your fork, so that the travis can finish.

I'll begin to take an actual look at this now.

**Comment by elken** [comment]
Ok, no problem. I tried to stick to the conventions you used, not sure if I met the indentation though. I'm just using the standard vim c++ syntax. 

Ik a lot of it is pretty sloppy, I'd be happy to clean it up if you guys are unhappy with anything.

**Comment by lolzballs** [comment]
RIP GIT!
image

**Comment by elken** [comment]
That's beautiful.

I call it "1am, no caffeine." by Damien Hurst

**Comment by lolzballs** [comment]
We have a crap load of those.
https://i.imgur.com/3hTk6dM.png
https://i.imgur.com/PODKQIA.png
https://i.imgur.com/6ZGMJls.png

**Comment by elken** [comment]
We can open up a gallery of post-modern art.

**Comment by lolzballs** [comment]
Works fine on my Ubuntu 15.04 machine.

There are some coding style changes that need to be fixed. `if`, `while`, `for`, etc. loops need to be followed with a space like the following: `if (time > 1 && caffeine < 10)`.

Also, in Library.cpp on line 81, that _should_ be common across *nix, so __unix__ should be better than __linux__.

**Comment by elken** [comment]
Yeah I saw that and I thought I did but I guess not. 

Ok, I'm off to sleep so I'll do this first thing. If you find anything else wrong, feel free to drop it here. 

On 5 July 2015 01:47:24 BST, Benjamin Cheng notifications@github.com wrote:

> Works fine on my Ubuntu 15.04 machine.
> 
> There are some coding style changes that need to be fixed. `if`,
> `while`, `for`, etc. loops need to be followed with a space like the
> following: `if (time > 1 && caffeine < 10)`.
> 
> Also, in Library.cpp on line 81, that _should_ be common across *nix,
> so **unix** should be better than **linux**.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/Proj-Ascension/Client/pull/34#issuecomment-118567727

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by elken** [comment]
Is that right?

**Comment by lolzballs** [comment]
Do not add spaces in the inside of the brackets, basically f45f437 was wrong.

**Comment by elken** [comment]
My bad, misread what you meant.

**Comment by Hal9007** [comment]
Hey elken,

This looks ready to go. Note that since this a largely new feature, it might be subject to a good many changes in the coming weeks. Thanks for your addition, anyhow.

File: Source/Database.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ Game Database::getGameById(unsigned int id)
     return {id, name, path, exe};
 }
 
-Game Database::getGameByName(QString name)
+Game* Database::getGameByName(QString name)
```
Comment (lolzballs) [inline Source/Database.cpp:77]: Wait, any reason for making this a pointer (other than causing memory leaks)?

Comment (elken) [inline Source/Database.cpp:77]: It was a lazy hack to compare it to nullptr. I meant to make a TODO to
find a better way to do it.

https://github.com/Proj-Ascension/Client/blob/dev/Source/Library.cpp#L249

(I know it's god awful, I didn't know how to check for existence better)

Comment (lolzballs) [inline Source/Database.cpp:77]: Oh, that's fine then. We'll just have to remember to fix it. :P

Comment (elken) [inline Source/Database.cpp:77]: We can take it out, it was just a check to verify that the game you're adding isn't already in there. It doesn't break anything, and a better way should be found IMO.

Are you fine with taking it out? If so, I'll hack it up & send a pr.

**Pull request merged**